ros/lib/utils.py

```diff
@@ -185,3 +193,62 @@ def log_request(self, code='-', size='-'):
             return
         else:
             super().log_request(code, size)
+
+
+def generate_highlight_description(instance_type):
+
+    if instance_type not in INSTANCE_TYPES.keys():
+        # logging lookup failure on Prometheus
+        ec2_instance_lookup_failures.labels(reporter='API Events').inc()
+        return 'NA'
+
+    cpu_type = INSTANCE_TYPES[instance_type]['extra']['physicalProcessor']
+    num_vcpus = INSTANCE_TYPES[instance_type]['extra']['vcpu']
+    ram_gb = INSTANCE_TYPES[instance_type]['extra']['memory']
+    cloud_regions = INSTANCE_TYPES[instance_type]['extra']['regionCode']
+    cloud = 'AWS'  # Update when multi cloud provider support is added
+
+    description_text = f'{cpu_type} instance with {num_vcpus} vCPUs ' \
+                       f'and {ram_gb} of RAM, running on {cloud} ' \
+                       f'{cloud_regions} regions'
+
+    return description_text
+
+
+def highlights_instance_types(queryset, highlight_type='historical'):
```

Comment:
@saltgen I feel it will be good if we can remove the `historical` as the default value for this function and make it accept 3 values to make it more readable, if you have any other suggestions for same do let me know 🙌🏻

Reply:
@PreetiW As discussed, we can do some minor refactoring on this one, I'll get to it once I'm back at work, thanks!